File: Examples/VisualBasic/WorkingWithTasks/ReadTaskPercentageCompletion.vb

Namespace WorkingWithTasks
    Public Class ReadTaskPercentageCompletion
        Public Shared Sub Run()
            ' ExStart:ReadTaskPercentageCompletion
            ' Create project instance
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)
            Dim project1 As New Project(dataDir + "project.mpp")

            ' Access tasks and display percentage completion
            Dim tasks = project1.RootTask.Children
            For Each tsk1 As Task In tasks
                Console.WriteLine(tsk1.Get(Tsk.PercentComplete))
                Console.WriteLine(tsk1.Get(Tsk.PercentWorkComplete).ToString())
                Console.WriteLine(tsk1.Get(Tsk.PhysicalPercentComplete).ToString())
            Next
            ' ExEnd:ReadTaskPercentageCompletion
        End Sub
    End Class
End Namespace

File: Examples/CSharp/WorkingWithTasks/ReadTaskPercentageCompletion.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithTasks
{
    using System;

    internal class ReadTaskPercentageCompletion
    {
        public static void Run()
        {
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod());
            
            //ExStart:ReadTaskPercentageCompletion
            //ExFor: Tsk.PercentComplete
            //ExFor: Tsk.PercentWorkComplete
            //ExFor: Tsk.PhysicalPercentComplete
            //ExSummary: Shows how to read task's percentages.
            var project = new Project(dataDir + "TaskPercentageCompletion.mpp");

            // Access tasks and display percentage completion
            var tasks = project.RootTask.Children;
            foreach (var task in tasks)
            {
                Console.WriteLine(task.Get(Tsk.PercentComplete));
                Console.WriteLine(task.Get(Tsk.PercentWorkComplete));
                Console.WriteLine(task.Get(Tsk.PhysicalPercentComplete));
            }
            //ExEnd:ReadTaskPercentageCompletion
        }
    }
}
